transportable-udfs-trino/src/main/java/com/linkedin/transport/trino/utils/TrinoKeywordsConverter.java

```diff
@@ -0,0 +1,128 @@
+          .put("CROSS", Boolean.FALSE)
+          .put("CUBE", Boolean.TRUE)
+          .put("CURRENT_DATE", Boolean.FALSE)
+          .put("CURRENT_TIME", Boolean.FALSE)
+          .put("CURRENT_TIMESTAMP", Boolean.FALSE)
+          .put("CURRENT_USER", Boolean.TRUE)
+          .put("DEALLOCATE", Boolean.FALSE)
+          .put("DELETE", Boolean.FALSE)
+          .put("DESCRIBE", Boolean.FALSE)
+          .put("DISTINCT", Boolean.FALSE)
+          .put("DROP", Boolean.FALSE)
+          .put("ELSE", Boolean.FALSE)
+          .put("END", Boolean.FALSE)
+          .put("ESCAPE", Boolean.FALSE)
+          .put("EXCEPT", Boolean.FALSE)
+          .put("EXECUTE", Boolean.FALSE)
+          .put("EXISTS", Boolean.FALSE)
+          .put("EXTRACT", Boolean.FALSE)
+          .put("FALSE", Boolean.FALSE)
+          .put("FOR", Boolean.FALSE)
+          .put("FROM", Boolean.FALSE)
+          .put("FULL", Boolean.FALSE)
+          .put("GROUP", Boolean.FALSE)
+          .put("GROUPING", Boolean.TRUE)
+          .put("HAVING", Boolean.FALSE)
+          .put("IN", Boolean.FALSE)
+          .put("INNER", Boolean.FALSE)
+          .put("INSERT", Boolean.FALSE)
+          .put("INTERSECT", Boolean.FALSE)
+          .put("INTO", Boolean.FALSE)
+          .put("IS", Boolean.FALSE)
+          .put("JOIN", Boolean.FALSE)
+          .put("LEFT", Boolean.FALSE)
+          .put("LIKE", Boolean.FALSE)
+          .put("LOCALTIME", Boolean.TRUE)
+          .put("LOCALTIMESTAMP", Boolean.TRUE)
+          .put("NATURAL", Boolean.FALSE)
+          .put("NORMALIZE", Boolean.TRUE)
+          .put("NOT", Boolean.FALSE)
+          .put("NULL", Boolean.FALSE)
+          .put("ON", Boolean.FALSE)
+          .put("OR", Boolean.FALSE)
+          .put("ORDER", Boolean.FALSE)
+          .put("OUTER", Boolean.FALSE)
+          .put("PREPARE", Boolean.FALSE)
+          .put("RECURSIVE", Boolean.TRUE)
+          .put("RIGHT", Boolean.FALSE)
+          .put("ROLLUP", Boolean.TRUE)
+          .put("SELECT", Boolean.FALSE)
+          .put("TABLE", Boolean.FALSE)
+          .put("THEN", Boolean.FALSE)
+          .put("TRUE", Boolean.FALSE)
+          .put("UESCAPE", Boolean.TRUE)
+          .put("UNION", Boolean.FALSE)
+          .put("UNNEST", Boolean.TRUE)
+          .put("USING", Boolean.FALSE)
+          .put("VALUES", Boolean.FALSE)
+          .put("WHEN", Boolean.FALSE)
+          .put("WHERE", Boolean.FALSE)
+          .put("WITH", Boolean.FALSE)
+          .build();
+
+  /**
+   * Quote the preserved keywords which might appear as field names in the type signatures
+   * i.e. `row(key varchar, values varchar)` will be converted to `row(key varchar, "values" varchar)`
+   * if they are not quoted, `io.trino.sql.analyzer.TypeSignatureTranslator.parseTypeSignature` will throw exception
+   *
+   * @param signature input type signature
+   * @return converted type signature which is properly quoted
+   */
+  public static String quoteReservedKeywordsInTypeSignature(String signature) {
```

Comment:
I wonder if using the regex pattern match is overkill?  We could have just searched the existence of a reserved word and replaced it with a quoted version?

Reply:
Replacing without matching using regex might fail for some corner cases like `row(valuesss varchar)`

Reply:
Yeah, I was thinking sth like tokenizing it using `String.split()`. I think I'm fine with your approach. Thanks!